tests/subsys/llext/simple/src/test_llext_simple.c

```diff
@@ -490,17 +490,16 @@ ZTEST(llext, test_printk_exported)
 }
 
 /*
- * Ensure ext_syscall_fail is exported - as it is picked up by the syscall
- * build machinery - but points to NULL as it is not implemented.
+ * The syscalls test above verifies that custom syscalls defined by extensions
+ * are properly exported. Since `ext_syscalls.h` declares ext_syscall_fail, we
+ * know it is picked up by the syscall build machinery, but the implementation
+ * for it is missing. Make sure the exported symbol for it is NULL.
  */
 ZTEST(llext, test_ext_syscall_fail)
 {
 	const void * const esf_fn = LLEXT_FIND_BUILTIN_SYM(z_impl_ext_syscall_fail);
 
```

Comment:
asserting est_fn is not null should still be done here no?

Reply:
No, `llext_find_sym` returns pointers to actual functions/data, so it makes sense for `esf_fn` to be NULL. 

As it is currently done in `main`,  and _only in the case of unimplemented syscalls_, the pointer returned by `llext_find_sym` points to a pointer to NULL (which is what was being tested here). But I assume this was a mistake, since:
1. the stored address is not NULL, so LLEXT happily links the code
2. at runtime your call to `z_impl_foo()` jumps at the location of a `void *z_impl_foo = NULL`, which obviously contains 4 or 8 zeroes. 

With this PR, the stored address is NULL, `find_sym` returns that NULL (not "no symbol", but "that symbol is defined and its value is NULL"), and the link process fails (as it should, there is no implementation for that symbol after all).